libraries/radpdfprocessing/features/embedded-file-streams/embedded-file-streams.md

```diff
@@ -30,18 +32,43 @@ RadFixedDocument stores the integrated files in an **EmbeddedFilesCollection** a
 
 {{endregion}}
 
->note RadPdfProcessing  provides support for embedding of [ZUGFeRD](https://de.wikipedia.org/wiki/ZUGFeRD) (acronym for Zentraler User Guide des Forums elektronische Rechnung Deutschland) invoices.
+>caution **DuplicatedEmbeddedFileNameException** is thrown when adding an embedded file with a name that is already added to the collection.
+
+#### Attachments section in Adobe 
+![Embedded Files in a PDF document](images/embedded_files_0.png)
+
+### Creating an Embedded Electronic (ZUGFeRD) Invoice
+
+RadPdfProcessing provides support for embedding of [ZUGFeRD](https://de.wikipedia.org/wiki/ZUGFeRD) (acronym for Zentraler User Guide des Forums elektronische Rechnung Deutschland) invoices.
 
-#### **[C#] Add ZUGFeRD invoide**
+#### **[C#] Add ZUGFeRD invoice**
 
 {{region cs-radpdfprocessing-embedded-file-add-zugferd-invoice}}
 
-byte[] bytes = File.ReadAllBytes(@"..\zugferd-invoice.xml");
-document.EmbeddedFiles.AddZugferdInvoice(bytes);
+            RadFixedDocument document = new RadFixedDocument();
+            using (RadFixedDocumentEditor editor = new RadFixedDocumentEditor(document))
+            {
+                editor.CharacterProperties.TrySetFont(new System.Windows.Media.FontFamily("Calibri"));
+                editor.InsertRun("Hello PDF/A");
+            };
+            byte[] bytes = File.ReadAllBytes(@"..\..\zugferd-invoice.xml");
+            document.EmbeddedFiles.AddZugferdInvoice(bytes);
+
+            PdfFormatProvider provider = new PdfFormatProvider();
+            PdfExportSettings settings = new PdfExportSettings();
+            settings.ComplianceLevel = PdfComplianceLevel.PdfA3B;
+            settings.ShouldEmbedFonts = true;
+            provider.ExportSettings = settings; 
+            using (Stream output = File.OpenWrite("exportedInvoice.pdf"))
+            { 
+                provider.Export(document, output);
+            }
 
 {{endregion}}
 
->caution [Standard Fonts]({%slug radpdfprocessing-concepts-fonts%}) should be embedded in the document.
+>note Only a single XML invoice attachment is allowed in ZUGFeRD.
+
+>caution Please avoid using [Standard Fonts]({%slug radpdfprocessing-concepts-fonts%}) because they are not being embedded in the document. 
```

Comment:
It might be changed for more detailed info with something like:

To comply with the PDF/A-3B standard all the fonts in the documents should be embedded, so please avoid using [Standard Fonts]({%slug radpdfprocessing-concepts-fonts%}) because they are not being embedded in the document.